controllers/etcd_controller.go

```diff
@@ -125,30 +133,16 @@ func (r *EtcdReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 				}, err
 			}
 		}
-		r.Logger.Infof("Deleted etcd %s successfully.", etcd.GetName())
+		logger.Infof("Deleted etcd %s successfully.", etcd.GetName())
 		return ctrl.Result{}, nil
 	}
 
 	// Add Finalizers to Etcd
-	if etcd.DeletionTimestamp.IsZero() {
-		if finalizers := sets.NewString(etcd.Finalizers...); !finalizers.Has(FinalizerName) {
-			r.Logger.Infof("Adding finalizer (%s) to etcd %s", FinalizerName, etcd.GetName())
-			finalizers.Insert(FinalizerName)
-			etcdCopy.Finalizers = finalizers.UnsortedList()
-			if err := r.Update(context.TODO(), etcdCopy); err != nil {
-				if err := r.updateEtcdErrorStatus(etcdCopy, err); err != nil {
-					return ctrl.Result{
-						Requeue:      true,
-						RequeueAfter: time.Second * 5,
-					}, nil
-				}
-				return ctrl.Result{
-					Requeue:      true,
-					RequeueAfter: time.Second * 5,
-				}, err
-			}
-		}
-		if err := r.addFinalizersToDependantSecrets(etcdCopy); err != nil {
+	if finalizers := sets.NewString(etcd.Finalizers...); !finalizers.Has(FinalizerName) {
```

Comment:
Do we need the finalizers?

Reply:
Yes. We need it. It's recommended that we have `finazliers` on the resource being acted on.  Basically without `finalizer` is somebody deletes the secret,  etcd-backup-restore pod will fail continuously.

Reply:
> without finalizer is somebody deletes the secret, etcd-backup-restore pod will fail continuously.

This is true for the secrets. But the finalizer above is for the `etcd` resource. Do we need that?

Reply:
Yes. Cause its definitely react on etcd resource. If etcd resource is deleted, then it has to do cleanup job like bring down the etcd sts.